monkey/monkey_island/cc/services/reporting/issue_processing/exploit_processing/exploiter_descriptor_enum.py

```diff
@@ -0,0 +1,32 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Type
+
+from monkey_island.cc.services.reporting.issue_processing.exploit_processing import CredExploitProcessor
+from monkey_island.cc.services.reporting.issue_processing.exploit_processing import ExploitProcessor
+from monkey_island.cc.services.reporting.issue_processing.exploit_processing import \
+    ShellShockExploitProcessor
+
+
+@dataclass
+class ExploiterDescriptor:
+    # Must match with class names of exploiters in Infection Monkey code
+    class_name: str
+    display_name: str
+    processor: Type[ExploitProcessor]
+
+
+class ExploiterDescriptorEnum(Enum):
+    SMB = ExploiterDescriptor('SmbExploiter', 'SMB Exploiter', CredExploitProcessor)
+    WMI = ExploiterDescriptor('WmiExploiter', 'WMI Exploiter', CredExploitProcessor)
+    SSH = ExploiterDescriptor('SSHExploiter', 'SSH Exploiter', CredExploitProcessor)
+    SAMBACRY = ExploiterDescriptor('SambaCryExploiter', 'SambaCry Exploiter', CredExploitProcessor)
+    ELASTIC = ExploiterDescriptor('ElasticGroovyExploiter', 'Elastic Groovy Exploiter', ExploitProcessor)
+    MS08_067 = ExploiterDescriptor('Ms08_067_Exploiter', 'Conficker Exploiter', ExploitProcessor)
+    SHELLSHOCK = ExploiterDescriptor('ShellShockExploiter', 'ShellShock Exploiter', ShellShockExploitProcessor)
+    STRUTS2 = ExploiterDescriptor('Struts2Exploiter', 'Struts2 Exploiter', ExploitProcessor)
+    WEBLOGIC = ExploiterDescriptor('WebLogicExploiter', 'Oracle WebLogic Exploiter', ExploitProcessor)
+    HADOOP = ExploiterDescriptor('HadoopExploiter', 'Hadoop/Yarn Exploiter', ExploitProcessor)
+    MSSQL = ExploiterDescriptor('MSSQLExploiter', 'MSSQL Exploiter', ExploitProcessor)
+    VSFTPD = ExploiterDescriptor('VSFTPDExploiter', 'VSFTPD Backdoor Exploiter', CredExploitProcessor)
+    DRUPAL = ExploiterDescriptor('DrupalExploiter', 'Drupal Server Exploiter', ExploitProcessor)
```

Comment:
The reason this isn't an actual link is because you don't want to include Monkey code in Island? Or why?

Reply:
Yes, because the island doesn't contain the agent's code.